### DISCUSSION TITLE
Strange error when I try to define my own operator

### DISCUSSION THREAD

**Discussion by xuyuon** [MilesCranmer/PySR#403] [Q&A] [answered]
I was trying to add to the model an unary operator to return a value from a Gaussian distribution, so I coded something like this:
```
model=PySRRegressor(
...
unary_operators=[
        "exp",
        "gau(x) = exp(-0.5 * x^2.0)",
    ],
...
```
but this would raise a runtime error: 
`TypeError: <PyCall.jlwrap (in a Julia function called from Python)
JULIA: TypeError: in typeassert, expected Float32, got a value of type Float64`

Therefore, I changed my code to:
```
model=PySRRegressor(
...
unary_operators=[
        "exp",
        "gau(x::Float64) = exp(-0.5 * x^2.0)",
    ],
...
```
This produced another error:
`RuntimeError: <PyCall.jlwrap (in a Julia function called from Python)
JULIA: AssertionError: Your configuration is invalid - one of your operators (gau) is not well-defined over the real line. You can get around this by returning `NaN` for invalid inputs.`

This error makes no sense because the function I wrote should be able to accept any real numbers. Anyway, I altered my code again:
```
model=PySRRegressor(
...
unary_operators=[
        "exp",
        "gau(x::Float64) = isfinite(x) ? exp(-0.5 * x^2.0) : NaN",
    ],
...
```
It was yet giving the same error:
`RuntimeError: <PyCall.jlwrap (in a Julia function called from Python)
JULIA: AssertionError: Your configuration is invalid - one of your operators (gau) is not well-defined over the real line. You can get around this by returning `NaN` for invalid inputs.`

I am stuck at this point, does anyone have some advice on this? Thanks in advance!

**Comment by MilesCranmer** [comment, marked as answer]
If you are using `precision=32` (default), then your operators have to use only `Float32`. So either use `precision=64`, or write your operator like this:

```
        "gau(x::T) where {T} = isfinite(x) ? exp(-T(0.5) * x^T(2)) : T(NaN)",
```

**Reply by MilesCranmer** [reply]
It might be nice if I convert the output automatically though... will see how hard this is to set up.

**Reply by xuyuon** [reply]
It works! Thank you very much!
